crates/virtio-device/src/mmio.rs

```diff
@@ -69,30 +74,32 @@ pub trait VirtioMmioDevice: WithDriverSelect {
         match offset {
             // The standard specifies that accesses to configuration registers are 32-bit wide.
             0x00..=0xff if data.len() == 4 => {
-                let v = match offset {
-                    0x0 => MMIO_MAGIC_VALUE,
-                    0x04 => MMIO_VERSION,
-                    0x08 => self.device_type(),
-                    0x0c => VENDOR_ID,
-                    0x10 => match self.device_features_select() {
+                let v = match offset as u32 {
```

Comment:
A good explanation here is that we are not loosing any important information if the driver sent a value that is more than the accepted u32 because in that case "undefined behavior" is expected.

Reply:
This is not needed. if we got here, we know that offset is between 0x00 and 0xff which is less than u32::max.